### PR TITLE
Add Forged Alliance Forever

### PR DESCRIPTION
**Pull request by chayleaf**
The client doesn't even start when not using the latest version. However, I don't want to maintain this package, updating the hash a few times a month. I hope you have some infrastructure ready for updating the hashes, if not, you can use something like [nvfetcher](https://github.com/berberman/nvfetcher) or [nix-update](https://github.com/Mic92/nix-update)

### CODE REVIEW

File: pkgs/faf-client/default.nix
Comment (fufexan) [inline pkgs/faf-client/default.nix:47]:
```suggestion
  stdenvNoCC.mkDerivation {
```
If we're not using the C compiler, we might as well use this.

File: pkgs/faf-client/default.nix
Comment (fufexan) [inline pkgs/faf-client/default.nix:39]: This should be sorted alphabetically.

File: pkgs/faf-client/default.nix
Comment (fufexan) [inline pkgs/faf-client/default.nix:25]: Is this subject to change? We could name it `faf-client-${sha256}.png` to avoid any name clashes, should the hash change in the future.

**Review by fufexan — CHANGES_REQUESTED** [review]
The package itself could use a bit more spacing between lines, just like packages in Nixpkgs.

**Comment by fufexan** [comment]
Thanks for the contribution!

I'm currently working on https://github.com/fufexan/nix-gaming/pull/43, so we will have automatic updating soon. The only blocker is an update script I have to write for osu-lazer-bin.

I will merge this PR after I finish that one (hopefully today).

File: pkgs/faf-client/default.nix
Comment (chayleaf) [reply pkgs/faf-client/default.nix:37]: actually, the desktop file uses the store path for the icon, so it should be fine

**Comment by fufexan** [comment]
Ok I've got npins sorted out. Now if you can add the update script to the `.github/workflows/update.yml` file, that's probably the last thing you need to do for this PR.

**Comment by chayleaf** [comment]
Just added `faf-client-setup` to simplify the install process for new users

File: pkgs/faf-client/default.nix
Comment (fufexan) [inline pkgs/faf-client/default.nix:31]: You should add yourself as a maintainer here.

Comment (chayleaf) [reply pkgs/faf-client/default.nix:31]: I will add myself when I get a first commit to nixpkgs

**Comment by fufexan** [comment]
If this is ready, I'll merge it when the checks pass. Otherwise ping me when it's ready.

**Comment by chayleaf** [comment]
yes, this is ready now @fufexan

File: pkgs/faf-client/update.sh
Comment (fufexan) [inline pkgs/faf-client/update.sh:2]: This will error out as there isn't a `NIX_PATH` set. You can either set `NIX_PATH=flake:nixpkgs=nixpkgs` which will pick it up from the registry, or use `nix shell`-style shebang, like this:
```sh
#!/usr/bin/env -S nix shell nixpkgs#curl nixpkgs#jq nixpkgs#gnugrep nixpkgs#gnused -c bash
```

Comment (chayleaf) [reply pkgs/faf-client/update.sh:2]: can you tell me which situations it will error out in? it works on my machine™

Comment (fufexan) [reply pkgs/faf-client/update.sh]: Right, it also works on mine. But in a GH runner it will fail with this:
```nix
Run pkgs/osu-lazer-bin/update.sh
error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

       at «string»:1:25:

            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (jq) (moreutils) ]; } ""
             |                         ^
(use '--show-trace' to show detailed location information)
```

**Review by fufexan — APPROVED** [review]
LGTM

**Pull request merged**
